jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -101,7 +102,9 @@ protected void doStart() throws Exception
         _selectorManager.execute(_strategy::produce);
 
         // Set started only if we really are started
-        submit(s->_started.set(true));
+        Start start = new Start();
+        submit(start);
+        start._started.await();
```

Comment:
I'm a bit cautious about this wait forever.... but I guess it is OK?

Reply:
Also `doStop()` is coded with `await()` so yes, I think it's OK.